### DISCUSSION TITLE
How do I upgrade a productive OpenSlides docker environment to a new version without loosing data?

### DISCUSSION THREAD

**Discussion by stebla27** [OpenSlides/OpenSlides#6254] [Q&A] [answered]
Hello,

I have an OpenSlides 3.3 docker environment up and running.
Now I want to upgrade to 3.4 and I have already compiled and uploaded the docker images to the registry?
But if I use docker compose now, all my data will be gone! Am I right?
How can I backup all files, settings, user,... and restore it after updating? Are there any instructions for this?
Many thanks in advance!

**Comment by tsiegleauq** [comment, marked as answer]
That roughly depends on your database which you used OpenSlides 3.3
If it was an external postgres, as we recommended for 3.3, nothing really changes. If you used the sqlite file you would need to make a copy first, move the sqlite into OS3.4 and start it to make a migration

**Reply by as-med** [reply]
Sorry for hijacking this quiet old discussion. @tsiegleauq 

We have an old installation of OpenSlides Version 3.3 installed from python packages and we want to migrate to version 4.x and I would like to use the export function introduced in version 3.4 for the data.
I already got docker container with version 3.4 running, but I'm unable to trigger the migration.

Where do I have to move the sqlite in OS3.4? Do I have to run any commands or is OpenSlides automatically starting the migration?
Do I have to change any settings in the settings.py?

Thanks a lot for a response.
I'm already trying for some days.
